torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -64,6 +65,32 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         prim::RequiresGradCheck /* (...)  -> (..., bool) */,
+         [](const Node* node) -> Operation {
+           std::vector<c10::optional<bool>> rg_props =
+               fmap(node->tys(attr::types), [](const TypePtr& t) {
+                 return t->cast<TensorType>()->requiresGrad();
+               });
+           return [rg_props](Stack* stack) {
+             auto num_inputs = rg_props.size();
+             TORCH_INTERNAL_ASSERT(
+                 stack->size() >= num_inputs && num_inputs > 0);
```

Comment:
Any reason to check `num_inputs > 0` here?  Since this is a static property of the `RequiresGradCheck` op, let's catch it at construction time.  (I also have mixed feelings about `TORCH_INTERNAL_ASSERT` in the interpreter loop period, but I can set that aside for now ;-) )

Reply:
err, that was a copy-pasta from TypeCheck! I think I should update both.